checkov/dockerfile/checks/UserExists.py

```diff
@@ -6,15 +6,20 @@ class UserExists(BaseDockerfileCheck):
     def __init__(self):
         name = "Ensure that a user for the container has been created"
         id = "CKV_DOCKER_3"
-        supported_instructions = ["*"]
+        supported_instructions = ["CMD", "ENTRYPOINT", "USER"]
         categories = [CheckCategories.IAM]
         super().__init__(name=name, id=id, categories=categories, supported_instructions=supported_instructions)
 
     def scan_entity_conf(self, conf):
-        for instruction, content in conf.items():
-            if instruction == "USER":
-                return CheckResult.PASSED, conf[instruction][0]
-        return CheckResult.FAILED, None
+        for instruction in conf:
+            value = instruction['value']
+            if instruction['instruction'] in self.supported_instructions:
```

Comment:
I'm surprised you need to check here, that the `instruction` is one of the supported ones, but maybe my memories are blurry 😄

Reply:
I copied this from another test case (`UpdateNotAlone.py`)